vm/src/vm/vm_core.rs

```diff
@@ -1012,6 +1012,24 @@ impl VirtualMachine {
         self.segments.memory.add_relocation_rule(src_ptr, dst_ptr)
     }
 
+    /// Add a new relocation rule, allowing for dst to be a `MaybeRelocatable`.
+    ///
+    /// Relocating memory to anything other than a `Relocatable` is generally not useful, but it
+    /// does make the implementation consistent with the pythonic version. In most cases it is
+    /// advisable to use `add_relocation_rule()`, which only accepts a Relocatable.
+    ///
+    /// Will return an error if any of the following conditions are not met:
+    ///   - Source address's segment must be negative (temporary).
+    ///   - Source address's offset must be zero.
+    ///   - There shouldn't already be relocation at the source segment.
```

Comment:
Shouldn't or mustn't? If the former, in which scenario would it be allowed? If the latter, can this function check it and return an error?

Reply:
"Mustn't" would be more appropriate here, although this verbiage was copied verbatim (it's already used a couple times in the codebase), so I hesitate to make it different.

This condition is checked in `fn Memory::add_relocation_rule_maybe_relocatable()`, which this fn calls: https://github.com/lambdaclass/cairo-vm/blob/159f67da19964cc54a95423a69470a26e534a13d/vm/src/vm/vm_memory/memory.rs#L338